packages/aws-cdk-lib/aws-dynamodb/lib/table-v2.ts

```diff
@@ -148,11 +149,19 @@ export interface TableOptionsV2 {
 
   /**
    * Whether point-in-time recovery is enabled.
-   *
-   * @default false
+   * @deprecated use `pointInTimeRecoverySpecification` instead
+   * @default false - point in time recovery is not enabled.
    */
   readonly pointInTimeRecovery?: boolean;
 
+  /**
+   * Whether point-in-time recovery is enabled
+   * and recoveryPeriodInDays is set.
+   *
+   * @default - point in time recovery is not enabled.
+   */
+  readonly pointInTimeRecoverySpecification?: PointInTimeRecoverySpecification;
```

Comment:
```suggestion
  readonly pointInTimeRecoveryPeriod?: Duration;
```
We should follow the [design guidelines](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#flat) and provide a flat interface:
1. We can avoid deprecating the `pointInTimeRecovery` property and build the `PointInTimeRecoverySpecification` CFN prop accordingly.
2. We should validate that `pointInTimeRecoveryPeriod` is between 1 and 35 days, and defaults to 35 ([docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-pointintimerecoveryspecification.html#cfn-dynamodb-table-pointintimerecoveryspecification-recoveryperiodindays)).
3. We should error if `pointInTimeRecovery` is explicitly set to `false` and `pointInTimeRecoveryPeriod` is set.

Reply:
Thanks for the review.

1. I've discussed with the team on the continuation of extra fields for DynamoDB and this if we continue to utilize a flat interface we would have a very long list of parameters. 
2. Do we need this validation, as CFN validates this for us and will throw the appropriate exception?
3. Same as above?

Reply:
Thanks @lpizzinidev for the detailed review. 

1. I had discussions with Lee on the design. While the design guidelines advocate for flattening nested properties to accommodate languages like Java, given that TypeScript and Python users account for approximately 95% of all CDK users, it might be worth revisiting this guideline with the team.
In my view, flattening properties introduces unnecessary noise by adding all options to the top-level namespace, making it harder for users to identify the right property. For TypeScript and Python users, nested structures are not only idiomatic but also provide better organization and ease of use programmatically.

2. For validation, it's an ongoing debate as well within the CDK team as some thinks we should leave the validation to CFN side in case if the rules change and CDK will need to make code changes to account for it (and often we won't be notified immediately) while some thinks validation on CDK side can help raise the error faster without deploying the template. IMO, I prefer to do the validation as well so I agree with you on this point.

3. I agree with you for the above reason.